src/network-manager.js

```diff
@@ -0,0 +1,281 @@
+/**
+ * Network Manager.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+'use strict';
+
+const child_process = require('child_process');
+const Platform = require('./platform');
+
+function stripQuotes(str) {
+  return str.replace(/^'+/, '').replace(/'+$/, '');
+}
+
+/**
+ * Get the WAN mode and options.
+ *
+ * @returns {Object} {mode: 'static|dhcp|pppoe|...', options: {...}}
+ */
+function getWanMode() {
+  let mode = null;
+  const options = {};
+
+  switch (Platform.getOS()) {
+    case 'linux-openwrt': {
+      const proc = child_process.spawnSync(
+        'uci',
+        ['show', 'network.wan'],
+        {encoding: 'utf8'}
+      );
+      if (proc.status !== 0) {
+        return {mode, options};
+      }
+
+      for (const line of proc.stdout.split('\n')) {
+        if (line.indexOf('=') < 0) {
+          continue;
+        }
+
+        const [key, value] = line.split('=', 2).map((s) => stripQuotes(s));
+
+        // discard network.wan=interface
+        if (!key.startsWith('network.wan.')) {
+          continue;
+        }
+
+        const opt = key.split('network.wan.')[1];
+        if (opt === 'proto') {
+          mode = value;
+        } else {
+          options[opt] = value;
+        }
+      }
+
+      return {mode, options};
+    }
+    default:
+      return {mode, options};
+  }
+}
+
+/**
+ * Set the WAN mode and options.
+ *
+ * @param {string} mode - static, dhcp, pppoe, ...
+ * @param {Object?} options - options specific to WAN mode
+ * @returns {boolean} Boolean indicating success.
+ */
+function setWanMode(mode, options = {}) {
+  const valid = ['static', 'dhcp', 'pppoe'];
+  if (!valid.includes(mode)) {
+    return false;
+  }
+
+  switch (Platform.getOS()) {
+    case 'linux-openwrt': {
+      let proc = child_process.spawnSync(
+        'uci',
+        ['set', `network.wan.proto=${mode}`]
+      );
+      if (proc.status !== 0) {
+        return false;
+      }
+
+      for (const [key, value] of Object.entries(options)) {
+        proc = child_process.spawnSync(
+          'uci',
+          ['set', `network.wan.${key}=${value}`]
+        );
+        if (proc.status !== 0) {
+          return false;
+        }
+      }
+
+      proc = child_process.spawnSync('uci', ['commit', 'network']);
+      if (proc.status !== 0) {
+        return false;
+      }
+
+      proc = child_process.spawnSync('/etc/init.d/network', ['reload']);
```

Comment:
This should be `restart` not `reload`

Reply:
The OpenWrt wiki disagrees: https://oldwiki.archive.openwrt.org/doc/uci/network

Reply:
ok - I take it back. Running `/etc/init.d/network help` shows:
```
Syntax: /etc/init.d/network [command]

Available commands:
        start   Start the service
        stop    Stop the service
        restart Restart the service
        reload  Reload configuration files (or restart if service does not implement reload)
        enable  Enable service autostart
        disable Disable service autostart
```
and if I change the SSID then using reload does in fact change the SSID being advertised.

I guess I've just been used to using restart from the regular system V init scripts.